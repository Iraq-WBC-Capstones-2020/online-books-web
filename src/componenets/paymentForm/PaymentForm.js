import React from 'react';

export default function PaymentForm() {
  return <div></div>;
}
