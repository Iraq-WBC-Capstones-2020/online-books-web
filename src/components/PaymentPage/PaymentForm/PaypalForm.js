import React from 'react';
import { MDBContainer, MDBRow, MDBCol, MDBInput, MDBBtn } from 'mdbreact';
import { useTranslation } from 'react-i18next';

export default function PaypalForm() {
  const { t } = useTranslation();
  return (
    <MDBContainer>
      <MDBRow>
        <MDBCol md="12">
          <form>
            <div className="grey-text">
              <MDBInput
                label={t('paymentPage.email')}
                icon="envelope"
                containerClass="d-flex flex-row-reverse"
                className="ml-0 mr-1"
                labelClass="ml-4.5"
                group
                type="email"
                validate
                error="wrong"
                success="right"
              />
              <MDBInput
                label={t('paymentPage.password')}
                icon="lock"
                containerClass="d-flex flex-row-reverse"
                className="ml-0 mr-1"
                labelClass="ml-4.5"
                group
                type="password"
                validate
              />
            </div>
            <div className="text-center mb-3">
              <MDBBtn
                type="button"
                color="blue"
                className="rounded-pill text-white z-depth-1a w-48"
              >
                {t('paymentPage.purchase')}
              </MDBBtn>
            </div>
          </form>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
}
