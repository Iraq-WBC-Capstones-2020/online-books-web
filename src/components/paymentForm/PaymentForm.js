import React from 'react';
import '../../assets/main.css';
import {useTranslation} from 'react-i18next';
import { MDBContainer, MDBRow, MDBCol, MDBIcon, MDBInput } from 'mdbreact';

export default function PaymentForm(){
  const { t } = useTranslation()
  return (
    <MDBContainer>
      <MDBRow>
        <MDBCol md="4">
          <form>
            <p className="h5 mx-5 my-5 text-gray-800">
              {t('PaymentPage.paymentMethod')}
            </p>
            <MDBRow>
              <MDBCol md="6">
                <p className="h6 text-center mx-3 text-gray-800 mb-4">
                  <MDBIcon className="mx-2 my-1" icon="dot-circle" />
                  {t('PaymentPage.creditCard')}
                </p>
              </MDBCol>
              <MDBCol md="6">
                <p className="h6 text-center mx-3 text-gray-800 mb-4">
                  <MDBIcon className="mx-2 my-1" far icon="circle" />
                  {t('PaymentPage.paypal')}
                </p>
              </MDBCol>
            </MDBRow>
            <div className="text-gray-600">
              <MDBInput
                label={t('PaymentPage.name')}
                icon="user"
                containerClass="d-flex flex-row-reverse"
                className="ml-0 mr-1"
                labelClass="ml-4.5"
                group
                type="text"
                validate
                error="wrong"
                success="right"
              />

              <MDBInput
                label={t('PaymentPage.cardNumber')}
                icon="credit-card"
                containerClass="d-flex flex-row-reverse"
                className="ml-0 mr-1"
                labelClass="ml-4.5"
                group
                type="email"
                validate
                error="wrong"
                success="right"
              />
              <MDBRow>
                <MDBCol md="6">
                  <MDBInput
                    label={t('PaymentPage.MMYYYY')}
                    icon="calendar"
                    containerClass="d-flex flex-row-reverse"
                    className="ml-0 mr-1"
                    labelClass="ml-4.5"
                    group
                    type="text"
                    validate
                    error="wrong"
                    success="right"
                  />
                </MDBCol>
                <MDBCol md="6">
                  <MDBInput
                    label={t('PaymentPage.CVC')}
                    icon="lock"
                    containerClass="d-flex flex-row-reverse"
                    className="ml-0 mr-1"
                    labelClass="ml-4.5"
                    group
                    type="password"
                    validate
                  />
                </MDBCol>
              </MDBRow>
            </div>
            <div className="text-center">
              <button class="bg-primary hover:bg-blue-700 text-white  py-2 px-4 rounded-full">
                {t('PaymentPage.purchase')}
              </button>
            </div>
          </form>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
}
