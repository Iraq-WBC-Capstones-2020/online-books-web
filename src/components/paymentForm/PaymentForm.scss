.gray-text {
  color: #3a3a3a;
}
.main-color {
  color: #2f80ed;
}
