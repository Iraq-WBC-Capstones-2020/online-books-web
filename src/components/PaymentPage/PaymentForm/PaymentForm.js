import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import {
  MDBContainer,
  MDBRow,
  MDBCol,
  MDBInput,
  MDBBtn,
} from 'mdbreact';
import PaypalForm from './PaypalForm';

export default function PaymentForm() {
  const { t } = useTranslation();
  const [paymentMethod, setPaymentMethod] = useState('credit card');

  const handleChange = (e) => {
    setPaymentMethod(e.target.value);
  };

  return (
    <MDBContainer>
      <MDBRow>
        <MDBCol md="12">
          <p className="h5 mx-5 my-5 text-gray-800">
            {t('paymentPage.paymentMethod')}
          </p>
          <MDBRow>
            <MDBCol md="6">
              <input
                type="radio"
                id="credit card"
                value="credit card"
                onChange={handleChange}
                checked={paymentMethod === 'credit card'}
                className="h6 text-center ml-5 mr-2 text-gray-800 mb-4"
              />
              <label for="credit card">{t('paymentPage.creditCard')}</label>
            </MDBCol>
            <MDBCol md="6">
              <input
                type="radio"
                id="paypal"
                value="paypal"
                onChange={handleChange}
                checked={paymentMethod === 'paypal' && <PaypalForm />}
                className="h6 text-center ml-3 mr-2 text-gray-800 mb-4"
              />
              <label for="paypal">{t('paymentPage.paypal')}</label>
            </MDBCol>
          </MDBRow>
        </MDBCol>
      </MDBRow>
      <MDBRow>
        {paymentMethod === 'credit card' ? <CreditForm /> : null}
        {paymentMethod === 'paypal' ? <PaypalForm /> : null}
      </MDBRow>
    </MDBContainer>
  );
}

function CreditForm() {
  const { t } = useTranslation();
  return (
    <MDBContainer>
      <MDBRow>
        <MDBCol md="12">
          <form>
            <div className="text-gray-600">
              <MDBInput
                label={t('paymentPage.name')}
                icon="user"
                containerClass="d-flex flex-row-reverse"
                className="ml-0 mr-1"
                labelClass="ml-4.5"
                group
                type="text"
                validate
                error="wrong"
                success="right"
              />

              <MDBInput
                label={t('paymentPage.cardNumber')}
                icon="credit-card"
                containerClass="d-flex flex-row-reverse"
                className="ml-0 mr-1"
                labelClass="ml-4.5"
                group
                type="email"
                validate
                error="wrong"
                success="right"
              />
              <MDBRow>
                <MDBCol md="6">
                  <MDBInput
                    label={t('paymentPage.MMYYYY')}
                    icon="calendar"
                    containerClass="d-flex flex-row-reverse"
                    className="ml-0 mr-1"
                    labelClass="ml-4.5"
                    group
                    type="text"
                    validate
                    error="wrong"
                    success="right"
                  />
                </MDBCol>
                <MDBCol md="6">
                  <MDBInput
                    label={t('paymentPage.CVC')}
                    icon="lock"
                    containerClass="d-flex flex-row-reverse"
                    className="ml-0 mr-1"
                    labelClass="ml-4.5"
                    group
                    type="password"
                    validate
                  />
                </MDBCol>
              </MDBRow>
            </div>
            <div className="text-center mb-3">
              <MDBBtn
                type="button"
                color="blue"
                className="rounded-pill text-white z-depth-1a w-48"
              >
                {t('paymentPage.purchase')}
              </MDBBtn>
            </div>
          </form>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
}
