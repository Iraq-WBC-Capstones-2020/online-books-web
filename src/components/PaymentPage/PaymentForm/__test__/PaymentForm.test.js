import React from 'react';
import renderer from 'react-test-renderer';
import PaymentForm from './../PaymentForm';

test('PaymentForm snapshot', () => {
  const tree = renderer.create(<PaymentForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
