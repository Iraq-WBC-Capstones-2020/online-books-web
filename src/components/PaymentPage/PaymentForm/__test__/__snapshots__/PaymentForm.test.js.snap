// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentForm snapshot 1`] = `
<div
  className="container"
  data-test="container"
>
  <div
    className="row"
    data-test="row"
  >
    <div
      className="col-md-12"
      data-test="col"
    >
      <p
        className="h5 mx-5 my-5 text-gray-800"
      >
        paymentPage.paymentMethod
      </p>
      <div
        className="row"
        data-test="row"
      >
        <div
          className="col-md-6"
          data-test="col"
        >
          <input
            checked={true}
            className="h6 text-center ml-5 mr-1 text-gray-800 mb-4"
            id="credit-card"
            onChange={[Function]}
            type="radio"
            value="credit-card"
          />
          <label
            for="credit-card"
          >
            paymentPage.creditCard
          </label>
        </div>
        <div
          className="col-md-6"
          data-test="col"
        >
          <input
            checked={false}
            className="h6 text-center ml-5 mr-1 text-gray-800 mb-4"
            id="paypal"
            onChange={[Function]}
            type="radio"
            value="paypal"
          />
          <label
            for="paypal"
          >
            paymentPage.paypal
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
    data-test="row"
  >
    <div
      className="container"
      data-test="container"
    >
      <div
        className="row"
        data-test="row"
      >
        <div
          className="col"
          data-test="col"
        >
          <form>
            <div
              className="text-gray-600"
            >
              <div
                className="md-form form-group d-flex flex-row-reverse"
              >
                <i
                  className="fa fa-user prefix"
                  data-test="fa"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                />
                <input
                  aria-disabled={false}
                  className="form-control validate ml-0 mr-1"
                  data-test="input"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  type="text"
                  value=""
                />
                <label
                  aria-labelledby=""
                  className="ml-4.5"
                  data-error="wrong"
                  data-success="right"
                  id=""
                  onClick={[Function]}
                >
                  paymentPage.name
                </label>
              </div>
              <div
                className="md-form form-group d-flex flex-row-reverse"
              >
                <i
                  className="fa fa-credit-card prefix"
                  data-test="fa"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                />
                <input
                  aria-disabled={false}
                  className="form-control validate ml-0 mr-1"
                  data-test="input"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  type="email"
                  value=""
                />
                <label
                  aria-labelledby=""
                  className="ml-4.5"
                  data-error="wrong"
                  data-success="right"
                  id=""
                  onClick={[Function]}
                >
                  paymentPage.cardNumber
                </label>
              </div>
              <div
                className="row"
                data-test="row"
              >
                <div
                  className="col-md-6"
                  data-test="col"
                >
                  <div
                    className="md-form form-group d-flex flex-row-reverse"
                  >
                    <i
                      className="fa fa-calendar prefix"
                      data-test="fa"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    />
                    <input
                      aria-disabled={false}
                      className="form-control validate ml-0 mr-1"
                      data-test="input"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onInput={[Function]}
                      type="text"
                      value=""
                    />
                    <label
                      aria-labelledby=""
                      className="ml-4.5"
                      data-error="wrong"
                      data-success="right"
                      id=""
                      onClick={[Function]}
                    >
                      paymentPage.MMYYYY
                    </label>
                  </div>
                </div>
                <div
                  className="col-md-6"
                  data-test="col"
                >
                  <div
                    className="md-form form-group d-flex flex-row-reverse"
                  >
                    <i
                      className="fa fa-lock prefix"
                      data-test="fa"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    />
                    <input
                      aria-disabled={false}
                      className="form-control validate ml-0 mr-1"
                      data-test="input"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onInput={[Function]}
                      type="password"
                      value=""
                    />
                    <label
                      aria-labelledby=""
                      className="ml-4.5"
                      data-error=""
                      data-success=""
                      id=""
                      onClick={[Function]}
                    >
                      paymentPage.CVC
                    </label>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="text-center mb-3"
            >
              <button
                className="btn-blue btn Ripple-parent rounded-pill text-white z-depth-1a w-48"
                data-test="button"
                onMouseUp={[Function]}
                onTouchStart={[Function]}
                type="button"
              >
                paymentPage.purchase
                <div
                  className="Ripple "
                  data-test="waves"
                  style={
                    Object {
                      "height": "0px",
                      "left": "0px",
                      "top": "0px",
                      "width": "0px",
                    }
                  }
                />
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
